internal/charmstore/server.go

```diff
@@ -52,6 +59,16 @@ type ServerParams struct {
 	// refreshes of entities in the stats cache. If this left as the
 	// zero value then the default of 1 hour will be used.
 	StatsCacheMaxAge time.Duration
+
+	// MaxConcurrentHTTPRequests specifies a limit on the maximum
```

Comment:
+1, we should at least state that each http request uses one session if not stated otherwise.